src/rest/QuerySuggest.ts

```diff
@@ -66,9 +66,40 @@ export interface IQuerySuggestRequest {
    * Specifies the context to use for the request.
    */
   context?: IStringMap<any>;
+  /**
+   * Specifies the second level of origin of the request, typically the identifier of the selected tab from which the request originates.
+   */
+  tab?: string;
+  /**
+   * Specifies the third level of origin of the request, typically the URL of the page that linked to the search interface from which the request originates (e.g., in JavaScript, this would correspond to the `document.referrer` value).
+   */
+  referrer?: string;
+  /**
+   * The locale of the current user. Will typically match the "language" parameter that is used to perform standard queries.
+   */
   locale?: string;
+  /**
+   * Specifies whether to attempt to complete the last word of the current "q" parameter and boost the ranking score of the resulting expression so that it is returned as the first query suggestion.
+   */
+  enableWordCompletion?: boolean;
+  /**
+   * Specfies the actions history which represents the past actions a user made. It is generated by the page view script (https://github.com/coveo/coveo.analytics.js)
+   */
+  actionsHistory?: any[];
+  /**
+   * The tz database identifier of the time zone to use to correctly interpret dates in the query expression and result items.
+   */
+  timezone?: string;
+  /**
+   * A GUID representing the current user, who can be authenticated or anonymous. This GUID is normally generated by the usage analytics service and stored in a non-expiring browser cookie.
+   */
+  visitorId?: string;
+  /**
+   * Whether the current user is anonymous. This can be specified when configuring the {@link SearchEndpoint}.
+   */
+  isGuestUser?: boolean;
+
   autoCompleter?: string;
```

Comment:
What about those options, are they worth a description also?

Reply:
They are deprecated and only there for legacy reasons, so I'd prefer to not comment them.